contracts/BaseStrategy.sol

```diff
@@ -228,6 +236,16 @@ abstract contract BaseStrategy {
         require(msg.sender == keeper || msg.sender == strategist || msg.sender == governance(), "!authorized");
         _;
     }
+    modifier useChi() {
+        if (address(chiToken) == address(0)) {
+            _;
+            return;
+        }
+        uint256 gasStart = gasleft();
+        _;
+        uint256 gasSpent = 21000 + gasStart - gasleft() + 16 * msg.data.length;
```

Comment:
Actually, now that I think about it, there could be issues with this math, or at least it is not clear to me if a possible overflow issue could happen.

Would this be the correct way to describe the calculation? (Should not be liable to overflow)
```suggestion
        uint256 gasSpent = (gasStart - gasleft()) + 21000 + (16 * msg.data.length);
```